### PR TITLE
Run tomcat as non-root user

### PR DESCRIPTION
**Pull request by jeanpommier**
update based on PR https://github.com/georchestra/mapstore2-georchestra/pull/442

### CODE REVIEW

**Comment by jeanpommier** [comment]
@pierrejego I believe we can merge this PR.

We will need to document in the migration notes that the mapstore writable datadir needs to be writable by UID/GID 999 (for the docker version)

**Review by edevosc2c — COMMENTED** [review]
Could you allow changing the UID and GID at build time?

You can do it with something like this:
```
ARG USER_GID=999
ARG USER_UID=999
RUN addgroup --gid ${USER_GID} tomcat && \
    adduser --system  -u ${USER_UID} --gid ${USER_GID} --no-create-home tomcat && \
    chown -R ${USER_UID}:${USER_GID} /usr/local/tomcat
```

**Comment by jeanpommier** [comment]
Hi @edevosc2c . Can you explain why for ? 

The idea is to run the georchestra official docker image, right ? Since this applies at build time, it means that to get different settings one would still have a build a custom image. I don't see the added value

**Comment by pmauduit** [comment]
> Can you explain why for ?

it looks like because of https://github.com/georchestra/georchestra/issues/4071

**Comment by edevosc2c** [comment]
> Hi @edevosc2c . Can you explain why for ?
> 
> The idea is to run the georchestra official docker image, right ? Since this applies at build time, it means that to get different settings one would still have a build a custom image. I don't see the added value

It's a standard thing in docker images to allow anyone to set UID and GID at build time.
It doesn't hurt to add it.

**Comment by jeanpommier** [comment]
Hi,
Sorry I had left this PR untidy. I've updated the PR according to your suggestions @edevosc2c. 
Does it look OK to you ?

File: Dockerfile [hunk 1]
```diff
@@ -38,4 +46,5 @@ ENV TERM xterm
 # Necessary to execute tomcat and custom scripts
 ENTRYPOINT ["/docker-entrypoint.sh"]
 CMD ["catalina.sh", "run"]
+
```
Comment (jeanmi151) [inline Dockerfile:49]: Do you forget to add somewhere : `USER tomcat` ?

**Review by edevosc2c — APPROVED** [review]
good for merging

**Comment by edevosc2c** [comment]
We need to fix the CI before merge if possible.

**Comment by edevosc2c** [comment]
hey @jeanpommier can you rebase against master? this should fix the CI because geosolutions fixed it: https://github.com/georchestra/mapstore2-georchestra/commit/fe581a2ccc070ca433b2701cf914ac3fa70a4a36#diff-7c5b1aec3fd401c4634f1d2e303fa0b61fef81dd181322616a7058e833328a50R74

**Comment by jeanpommier** [comment]
> hey @jeanpommier can you rebase against master? this should fix the CI because geosolutions fixed it: [fe581a2#diff-7c5b1aec3fd401c4634f1d2e303fa0b61fef81dd181322616a7058e833328a50R74](https://github.com/georchestra/mapstore2-georchestra/commit/fe581a2ccc070ca433b2701cf914ac3fa70a4a36#diff-7c5b1aec3fd401c4634f1d2e303fa0b61fef81dd181322616a7058e833328a50R74)

Hi @edevosc2c 
Done, thanks

**Review by edevosc2c — APPROVED** [review]

**Comment by edevosc2c** [comment]
will probably be great to create a new mapstore release

**Comment by landryb** [comment]
there will be an rc for 2024.02 soon, see #714. unless you want to cut a release from the previous 'stable' branch.

**Comment by landryb** [comment]
actually there has been one in https://github.com/georchestra/mapstore2-georchestra/releases/tag/2024.02.00-RC1-geOrchestra

**Comment by edevosc2c** [comment]
Yes we want to backport that to previous versions. Some of our clients want this change.

**Pull request merged**